### ISSUE TITLE
Colors provided to plot color argument do not match patches

### ISSUE THREAD

**Issue by harmbuisman** [geopandas/geopandas#1075]
A geopandas dataframe has the option to plot the frame. This also has a color argument which can be provided a list of colors.

The colors do however not match the patches in each row, even though the list of colors provided is the same length as the number of rows in the geopandas data frame. 

Running the following code using the attached geojson file in [bug_df.zip](https://github.com/geopandas/geopandas/files/3451023/bug_df.zip) shows this :
```
import geopandas as gpd
import matplotlib.pyplot as plt
plot_df = gpd.read_file('bug_df.geojson')
plot_df['coords'] = plot_df['geometry'].apply(lambda x: \
                                    x.representative_point().coords[:][0])

fig, ax = plt.subplots(figsize=[15,15])
plot_df.plot(ax=ax, color=plot_df['color'])

for _, row in plot_df.iterrows():
    ax.annotate(s=row['color'], xy=row['coords'],
                horizontalalignment='center',
                color=row['color'])
plt.title('Wrong coloring')

fig, ax = plt.subplots(figsize=[15,15])

for col in plot_df['color'].unique():
    plot_df[plot_df['color']==col].plot(ax=ax, color=col)
plt.title('Correct coloring');
```

The output of this code shows that the colors for the patches do not match the expected color:
![image](https://user-images.githubusercontent.com/7702207/62202810-49693080-b38a-11e9-941f-c54472218887.png)

As a workaround (also in the code above) one can plot each of the patches separately with the right color. This gives the correct plot:
![image](https://user-images.githubusercontent.com/7702207/62202884-6c93e000-b38a-11e9-949b-42f6893078cc.png)

**Comment by martinfleis** [comment]
The list of colors is matched with the list of patches, not rows. If your row contains MultiPolygons, which they do, it will not match. A workaround is to explode your gdf and use `color` column of exploded gdf (or plot it directly). But we should probably look at it and copy the same color for all parts of multipolygon. I've seen related issue, but can't find it now.

```
plot_dfex = plot_df.explode()
fig, ax = plt.subplots(figsize=[15,15])
plot_df.plot(ax=ax, color=plot_dfex['color'])
```

![plot](https://user-images.githubusercontent.com/36797143/62900220-87cefa00-bd51-11e9-9250-8f3463cf957a.png)

**Comment by harmbuisman** [comment]
@martinfleis, thanks for the suggestion. I use the Bokeh package to visualize the plots and when I use explode disconnected land masses of the same region no longer seem to be connected. Not sure if that is another problem in itself though.

The list of colors matched to the patches itself would be a bit hard to manage for me. I have no idea how the externally downloaded shape file multipolygons are split into polygons. From my use case I just 'care' about the regions themselves, which is by gpd dataframe row which happens to contain a multipolygon in the geometry field.

I would be interested to hear the arguments for assigning the colors to the polygons. Perhaps there are use cases that I have not considered.

**Comment by martinfleis** [comment]
> I would be interested to hear the arguments for assigning the colors to the polygons. Perhaps there are use cases that I have not considered.

@harmbuisman That is a bug, not an intention. It is fixed in #1119, so you can try that specific branch until it will be merged.

**Comment by harmbuisman** [comment]
@martinfleis, very nice, thanks!
